ci/coverage

```diff
@@ -16,36 +16,43 @@
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
 
+set -o errexit
+set -o pipefail
+
 LCOV=${LCOV:-lcov}
 GENHTML=${GENHTML:-genhtml}
 
 TMPDIR=${TMPDIR:-/tmp}
 BUILDID="org.apache.trafficserver.$$"
 
-SRCROOT=${SRCROOT:-$(cd $(dirname $0)/.. && pwd)} # where the source lives
+SRCROOT=${SRCROOT:-$(cd "$(dirname "$0")/.." && pwd)} # where the source lives
 OBJROOT=${OBJROOT:-"$TMPDIR/$BUILDID/obj"} # where we are building
 DSTROOT=${DSTROOT:-"$TMPDIR/$BUILDID/dst"} # where we are installing
 
 # Force low make parallelization so that the build can complete in a VM with
 # only a small amount of memory.
 NPROCS=${NPROCS:-2}
 
-mkdir -p $SRCROOT
-mkdir -p $OBJROOT
-mkdir -p $DSTROOT
+mkdir -p "$SRCROOT"
+mkdir -p "$OBJROOT"
+mkdir -p "$DSTROOT"
 
 autogen() {
   (
     cd "$SRCROOT"
-    [ configure -nt configure.ac -a Makefile.in -nt Makefile.am ] || autoreconf -fi
+    if [ configure -nt configure.ac ] && [ Makefile.in -nt Makefile.am ] ; then
+        :
+    else
+        autoreconf -fi
+    fi
```

Comment:
I prefer the original.

If not the original, then at least let's reverse the boolean and not have the empty then.

Reply:
This change is based on https://www.shellcheck.net/wiki/SC2166.The work happens in the else because I didn't want to invert the conditional.

I do agree that it looks a bit weird :)